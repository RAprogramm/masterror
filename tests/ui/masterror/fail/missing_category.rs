// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::{AppCode, Masterror};

#[derive(Debug, Masterror)]
#[error("oops")]
#[masterror(code = AppCode::Internal)]
struct MissingCategory;

fn main() {}
