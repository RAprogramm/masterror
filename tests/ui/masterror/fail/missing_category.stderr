error: missing `category = ...` in #[masterror(...)]
 --> tests/ui/masterror/fail/missing_category.rs:5:1
  |
5 | #[masterror(code = AppCode::Internal)]
  | ^

warning: unused import: `AppCode`
 --> tests/ui/masterror/fail/missing_category.rs:1:17
  |
1 | use masterror::{AppCode, Masterror};
  |                 ^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
